provisionersdk/scripts/bootstrap_windows.ps1

```diff
@@ -1,9 +1,23 @@
-# On Windows, VS Code Remote requires a parent process of the
-# executing shell to be named "sshd", otherwise it fails. See:
-# https://github.com/microsoft/vscode-remote-release/issues/5699
-$ProgressPreference = "SilentlyContinue"
-Invoke-WebRequest -Uri ${ACCESS_URL}bin/coder-windows-${ARCH}.exe -OutFile $env:TEMP\sshd.exe
-Set-MpPreference -DisableRealtimeMonitoring $true -ExclusionPath $env:TEMP\sshd.exe
-$env:CODER_AGENT_AUTH = "${AUTH_TYPE}"
-$env:CODER_AGENT_URL = "${ACCESS_URL}"
-Start-Process -FilePath $env:TEMP\sshd.exe -ArgumentList "agent" -PassThru
+while ($true) {
+	try {
+		$ProgressPreference = "SilentlyContinue"
+		# On Windows, VS Code Remote requires a parent process of the
+		# executing shell to be named "sshd", otherwise it fails. See:
+		# https://github.com/microsoft/vscode-remote-release/issues/5699
+		$BINARY_URL="${ACCESS_URL}/bin/coder-windows-${ARCH}.exe"
+		Invoke-WebRequest -Uri "${BINARY_URL}" -OutFile $env:TEMP\sshd.exe
+		Set-MpPreference -DisableRealtimeMonitoring $true -ExclusionPath $env:TEMP\sshd.exe
+		$env:CODER_AGENT_AUTH = "${AUTH_TYPE}"
+		$env:CODER_AGENT_URL = "${ACCESS_URL}"
+		Start-Process -FilePath $env:TEMP\sshd.exe -ArgumentList "agent" -PassThru
```

Comment:
I'm not sure of the Windows specific behavior here, but it seems like we'll keep trying to re-download and start the agent even when the agent exits non-zero, is that what we want? (Asking because this behavior is different from the unix-y scripts.)

Reply:
Good point, I'll move that outside the loop to be consistent with the *nix scripts.